apps/api/src/planning-session/planning-session.service.ts

```diff
@@ -27,8 +33,28 @@ export class PlanningSessionService {
     private allowedActRepo: Repository<AllowedActivity>,
   ) {}
 
-  async createPlanningSession(): Promise<PlanningSession> {
-    const planningSession = this.planningSessionRepo.create();
+  // find latest Draft planning sessions
+  async getDraftPlanningSession(user: KeycloakUser): Promise<PlanningSession | undefined> {
+    const planningSession = await this.planningSessionRepo.findOne({
+      where: {
+        status: PlanningStatus.DRAFT,
+        createdBy: user.sub,
+      },
+      order: {
+        createdAt: -1,
+      },
+    });
+
+    return planningSession;
```

Comment:
Returning undefined on a miss also seems like an anti pattern.

Reply:
do you suggest, something like `data = {}`?

Reply:
actually, this is just a service, and not controller; Adding `{ data: {} }` to a controller makes more sense than to a service. What do you think?

Also, I am going ahead with the merge for now. And we can continue this thread here.